File: languages/typescript/Zone.ts
import { BaseElement } from './base_elements';

export interface Zone extends BaseElement {
  // Scope
  function?: string | null;
  startDate?: number | null;
  endDate?: number | null;
  phenomenaIds?: string[] | null;
  // World
  history?: string | null;
  claimedByIds?: string[] | null;
  roamedByIds?: string[] | null;
  titlesIds?: string[] | null;
}

File: languages/csharp_unity/Zone.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Zone : BaseElement
{
    // Scope
    public string Function;
    public int? StartDate;
    public int? EndDate;
    public List<string> PhenomenaIds; = new List<string>();
    // World
    public string History;
    public List<string> ClaimedByIds; = new List<string>();
    public List<string> RoamedByIds; = new List<string>();
    public List<string> TitlesIds; = new List<string>();
}
